langchain/prompts/few_shot.py

```diff
@@ -120,8 +140,73 @@ def _prompt_type(self) -> str:
         """Return the prompt type key."""
         return "few_shot"
 
-    def dict(self, **kwargs: Any) -> Dict:
-        """Return a dictionary of the prompt."""
-        if self.example_selector:
-            raise ValueError("Saving an example selector is not currently supported")
-        return super().dict(**kwargs)
+
+# For backwards compatibility.
+FewShotPromptTemplate = FewShotStringPromptTemplate
+
+
+class FewShotChatPromptTemplate(BaseChatPromptTemplate, BaseFewShotPromptTemplate):
+    """Prompt template that contains few shot examples."""
+
+    example_input: str
+    example_output: str
+    examples: List[Dict[str, str]]
+
+    @property
+    def _prompt_type(self) -> str:
+        """Return the prompt type key."""
+        return "few_shot_chat"
+
+    def format(self, **kwargs: Any) -> Sequence[BaseMessage]:
+        """Format to a sequence of BaseMessages."""
+
+        message_prompts = []
+        if self.prefix:
+            prefix_message_prompt = SystemMessagePromptTemplate(
+                prompt=StringPromptTemplate.from_template(self.prefix)
+            )
+            message_prompts.append(prefix_message_prompt)
+
+        # TODO: add support for example selectors
+        for example in self._get_examples():
+            message_prompts.append(
+                HumanMessagePromptTemplate(
+                    prompt=StringPromptTemplate.from_template(
+                        self.example_input.format(**example)
+                    )
+                ),
+            )
+            message_prompts.append(
+                AIMessagePromptTemplate(
+                    prompt=StringPromptTemplate.from_template(
+                        self.example_output.format(**example)
+                    )
+                ),
+            )
+
+        # construct the suffix message
+        suffix_message_prompt = HumanMessagePromptTemplate(
+            prompt=StringPromptTemplate.from_template(self.suffix)
+        )
+        message_prompts.append(suffix_message_prompt)
+
+        chat_prompt_template = ChatPromptTemplate.from_messages(message_prompts)
+        return chat_prompt_template.format(**kwargs)
+
+
+if __name__ == "__main__":
```

Comment:
shouldnt stay in

Reply:
yeah totally, just wanted to provide an example without spinning up a notebook